parsemsgpack.m

```diff
@@ -193,25 +194,25 @@ case uint8(223) % map32
 end
 
 function [out, idx] = parsemap(len, bytes, idx)
-%     out = containers.Map();
-%     for n=1:len
-%         [key, idx] = parse(bytes, idx);
-%         [out(string(key)), idx] = parse(bytes, idx);
-%     end
-
-    % In most of the cases using struct is faster and more user friendly.
-    % Choose how to manage maps by commenting out/in.
-
-    out = struct();
+    out = containers.Map();
     for n=1:len
         [key, idx] = parse(bytes, idx);
-        [out.(replaceMsgPackKey(key)), idx] = parse(bytes, idx);
```

Comment:
Frankly, we should just use `matlab.lang.makeValidName` instead of `replaceMsgPackKey`. I understand that `replaceMsgPackKey` is perhaps faster, but it is also unexpected behavior.

However, reading through the code, and realizing that MsgPack differs from JSON in that MsgPack Map keys are not constrained to strings, I think we should just drop the struct compatibility. I'd be fine with `matlab.lang.makeValidName` for string keys, but since they're not constrained to be strings, structs just don't make sense. Sorry.